common/image-fit.c

```diff
@@ -1224,6 +1224,9 @@ int calculate_hash(const void *data, int data_len, const char *algo,
 		sha1_csum_wd((unsigned char *)data, data_len,
 			     (unsigned char *)value, CHUNKSZ_SHA1);
 		*value_len = 20;
+	} else if (IMAGE_ENABLE_MD5 && strcmp(algo, "md5") == 0) {
+		md5_wd((unsigned char *)data, data_len, value, CHUNKSZ_MD5);
```

Comment:
Is this function defined when MD5 is not enabled/used?

Reply:
The same cases as for crc32/sha1. Look at lines 1216-1217.

Reply:
It was just protected with 2 different ifdef's because these algos crc32, sha1, and md5 were just non-contiguous :)

So this patch doesn't change logic at all, it just allows to remove extra ifdef/endif lines.

Reply:
No, this patch wont increase the size of the binary.